Prefer std::abs over fabs

LumioseSil opened the pull request:
#### da934454c84ac2dcbf9fca9e5f4ac2644ef25d72
<pre>
Prefer std::abs over fabs
<a href="https://bugs.webkit.org/show_bug.cgi?id=252974">https://bugs.webkit.org/show_bug.cgi?id=252974</a>

Reviewed by Ryosuke Niwa.

Unlike the C library versions, std::abs has overloads for every type, so one does not need to worry about which version of the function to call.

* Source/JavaScriptCore/assembler/MacroAssembler.h:
* Source/JavaScriptCore/b3/B3ConstDoubleValue.cpp:
* Source/JavaScriptCore/b3/B3ConstFloatValue.cpp:
* Source/JavaScriptCore/b3/testb3_3.cpp:
* Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h:
* Source/JavaScriptCore/dfg/DFGOperations.cpp:
* Source/JavaScriptCore/jit/AssemblyHelpersSpoolers.h:
* Source/JavaScriptCore/runtime/DateConversion.cpp:
* Source/JavaScriptCore/runtime/ISO8601.cpp:
* Source/JavaScriptCore/runtime/MathCommon.cpp:
* Source/JavaScriptCore/runtime/MathObject.cpp:
* Source/JavaScriptCore/runtime/NumberPrototype.cpp:
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
* Source/WTF/wtf/CurrentTime.cpp:
* Source/WTF/wtf/DateMath.cpp:
* Source/WTF/wtf/MediaTime.cpp:
* Source/WTF/wtf/text/TextStream.cpp:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
* Source/WebCore/Modules/webaudio/AudioParam.cpp:
* Source/WebCore/Modules/webaudio/AudioParamTimeline.cpp:
* Source/WebCore/Modules/webaudio/IIRFilterNode.cpp:
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
* Source/WebCore/Modules/webaudio/PeriodicWave.cpp:
* Source/WebCore/Modules/webaudio/RealtimeAnalyser.cpp:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
* Source/WebCore/bindings/js/JSDOMConvertNumbers.cpp:
* Source/WebCore/contentextensions/DFABytecodeCompiler.cpp:
* Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp:
* Source/WebCore/editing/FrameSelection.cpp:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/html/CustomPaintImage.cpp:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/inspector/InspectorOverlay.cpp:
* Source/WebCore/layout/Verification.cpp:
* Source/WebCore/page/EventHandler.cpp:
* Source/WebCore/page/Frame.cpp:
* Source/WebCore/page/WheelEventDeltaFilter.cpp:
* Source/WebCore/page/ios/FrameIOS.mm:
* Source/WebCore/platform/LayoutUnit.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
* Source/WebCore/platform/ScrollAnimationKeyboard.cpp:
* Source/WebCore/platform/ScrollAnimationKinetic.cpp:
* Source/WebCore/platform/ScrollView.cpp:
* Source/WebCore/platform/audio/Biquad.cpp:
* Source/WebCore/platform/audio/Cone.cpp:
* Source/WebCore/platform/audio/DenormalDisabler.h:
* Source/WebCore/platform/audio/FFTFrame.cpp:
* Source/WebCore/platform/audio/HRTFPanner.cpp:
* Source/WebCore/platform/audio/IIRFilter.cpp:
* Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm:
* Source/WebCore/platform/graphics/FloatLine.cpp:
* Source/WebCore/platform/graphics/FloatSize.cpp:
* Source/WebCore/platform/graphics/GradientImage.cpp:
* Source/WebCore/platform/graphics/IntSize.h:
* Source/WebCore/platform/graphics/PathUtilities.cpp:
* Source/WebCore/platform/graphics/ShadowBlur.cpp:
* Source/WebCore/platform/graphics/UnitBezier.h:
* Source/WebCore/platform/graphics/cairo/CairoOperations.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/TiledBackingStore.cpp:
* Source/WebCore/platform/graphics/transforms/AffineTransform.cpp:
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp:
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
* Source/WebCore/platform/graphics/win/SystemFontDatabaseWin.cpp:
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
* Source/WebCore/platform/ios/WebAVPlayerController.mm:
* Source/WebCore/platform/ios/wak/WKView.mm:
* Source/WebCore/platform/mac/ScrollAnimationRubberBand.mm:
* Source/WebCore/platform/mac/ScrollingEffectsController.mm:
* Source/WebCore/platform/win/BitmapInfo.h:
* Source/WebCore/platform/win/PopupMenuWin.cpp:
* Source/WebCore/platform/xr/openxr/OpenXRUtils.h:
* Source/WebCore/rendering/AttachmentLayout.mm:
* Source/WebCore/rendering/BorderPainter.cpp:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
* Source/WebCore/rendering/RenderListBox.cpp:
* Source/WebCore/rendering/RenderMarquee.cpp:
* Source/WebCore/rendering/shapes/RasterShape.cpp:
* Source/WebCore/rendering/svg/SVGMarkerData.h:
* Source/WebCore/rendering/svg/SVGTextLayoutEngineBaseline.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleFontSizeFunctions.cpp:
* Source/WebCore/svg/SVGPathParser.cpp:
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/gtk/KeyBindingTranslator.cpp:
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
* Source/WebKit/UIProcess/ios/WKKeyboardScrollingAnimator.mm:
* Source/WebKit/UIProcess/win/WebPopupMenuProxyWin.cpp:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
* Source/WebKitLegacy/ios/WebCoreSupport/WebFrameIOS.mm:
* Source/WebKitLegacy/mac/WebView/WebDynamicScrollBarsView.mm:
* Tools/TestRunnerShared/EventSerialization/mac/EventSerializerMac.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/GPUProcess.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
* Tools/TestWebKitAPI/Tests/ios/FullscreenTouchSecheuristicTests.cpp:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:

Canonical link: <a href="https://commits.webkit.org/261584@main">https://commits.webkit.org/261584@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9982a47ba33872c4fc6c67ca8d0cc2d259943d6d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112208 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21343 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/835 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4009 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120853 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116270 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22690 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/12426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/4775 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117971 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16858 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100041 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105270 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/98815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/88/builds/582 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/45869 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/100614 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13750 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/623 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/93865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/11873 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14428 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/12426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/101960 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19793 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52622 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31875 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8077 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16222 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/110003 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/27176 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rniwa reviewed (changes requested):
Commit message is in wrong format.

rniwa reviewed (changes requested):
The commit message needs to have a full list of files and functions like other commits.

rniwa commented on Source/JavaScriptCore/b3/testb3_3.cpp:
```diff
@@ -1812,7 +1812,7 @@ void testAbsAbsArg(float a)
     root->appendNewControlValue(proc, Return, Origin(), secondAbs);
 
-    CHECK(isIdentical(compileAndRun<float>(proc, bitwise_cast<int32_t>(a)), static_cast<float>(fabs(fabs(a)))));
+    CHECK(isIdentical(compileAndRun<float>(proc, bitwise_cast<int32_t>(a)), static_cast<float>(std::abs(std::abs(a)))));
```
What's the point of calling ads twice here?

AZero13 replied:
Beats me. This is a test though.

rniwa reviewed (approved):
r=me

LumioseSil committed:
Prefer std::abs over fabs
https://bugs.webkit.org/show_bug.cgi?id=252974

Reviewed by Ryosuke Niwa.

Unlike the C library versions, std::abs has overloads for every type, so one does not need to worry about which version of the function to call.

* Source/JavaScriptCore/assembler/MacroAssembler.h:
* Source/JavaScriptCore/b3/B3ConstDoubleValue.cpp:
* Source/JavaScriptCore/b3/B3ConstFloatValue.cpp:
* Source/JavaScriptCore/b3/testb3_3.cpp:
* Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h:
* Source/JavaScriptCore/dfg/DFGOperations.cpp:
* Source/JavaScriptCore/jit/AssemblyHelpersSpoolers.h:
* Source/JavaScriptCore/runtime/DateConversion.cpp:
* Source/JavaScriptCore/runtime/ISO8601.cpp:
* Source/JavaScriptCore/runtime/MathCommon.cpp:
* Source/JavaScriptCore/runtime/MathObject.cpp:
* Source/JavaScriptCore/runtime/NumberPrototype.cpp:
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
* Source/WTF/wtf/CurrentTime.cpp:
* Source/WTF/wtf/DateMath.cpp:
* Source/WTF/wtf/MediaTime.cpp:
* Source/WTF/wtf/text/TextStream.cpp:
* Source/WebCore/Modules/mediasession/MediaSessionCoordinator.cpp:
* Source/WebCore/Modules/webaudio/AudioParam.cpp:
* Source/WebCore/Modules/webaudio/AudioParamTimeline.cpp:
* Source/WebCore/Modules/webaudio/IIRFilterNode.cpp:
* Source/WebCore/Modules/webaudio/OscillatorNode.cpp:
* Source/WebCore/Modules/webaudio/PeriodicWave.cpp:
* Source/WebCore/Modules/webaudio/RealtimeAnalyser.cpp:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
* Source/WebCore/bindings/js/JSDOMConvertNumbers.cpp:
* Source/WebCore/contentextensions/DFABytecodeCompiler.cpp:
* Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp:
* Source/WebCore/editing/FrameSelection.cpp:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/html/CustomPaintImage.cpp:
* Source/WebCore/html/HTMLMediaElement.cpp:
* Source/WebCore/inspector/InspectorOverlay.cpp:
* Source/WebCore/layout/Verification.cpp:
* Source/WebCore/page/EventHandler.cpp:
* Source/WebCore/page/Frame.cpp:
* Source/WebCore/page/WheelEventDeltaFilter.cpp:
* Source/WebCore/page/ios/FrameIOS.mm:
* Source/WebCore/platform/LayoutUnit.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
* Source/WebCore/platform/ScrollAnimationKeyboard.cpp:
* Source/WebCore/platform/ScrollAnimationKinetic.cpp:
* Source/WebCore/platform/ScrollView.cpp:
* Source/WebCore/platform/audio/Biquad.cpp:
* Source/WebCore/platform/audio/Cone.cpp:
* Source/WebCore/platform/audio/DenormalDisabler.h:
* Source/WebCore/platform/audio/FFTFrame.cpp:
* Source/WebCore/platform/audio/HRTFPanner.cpp:
* Source/WebCore/platform/audio/IIRFilter.cpp:
* Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm:
* Source/WebCore/platform/graphics/FloatLine.cpp:
* Source/WebCore/platform/graphics/FloatSize.cpp:
* Source/WebCore/platform/graphics/GradientImage.cpp:
* Source/WebCore/platform/graphics/IntSize.h:
* Source/WebCore/platform/graphics/PathUtilities.cpp:
* Source/WebCore/platform/graphics/ShadowBlur.cpp:
* Source/WebCore/platform/graphics/UnitBezier.h:
* Source/WebCore/platform/graphics/cairo/CairoOperations.cpp:
* Source/WebCore/platform/graphics/texmap/coordinated/TiledBackingStore.cpp:
* Source/WebCore/platform/graphics/transforms/AffineTransform.cpp:
* Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp:
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
* Source/WebCore/platform/graphics/win/SystemFontDatabaseWin.cpp:
* Source/WebCore/platform/ios/ScrollAnimatorIOS.mm:
* Source/WebCore/platform/ios/WebAVPlayerController.mm:
* Source/WebCore/platform/ios/wak/WKView.mm:
* Source/WebCore/platform/mac/ScrollAnimationRubberBand.mm:
* Source/WebCore/platform/mac/ScrollingEffectsController.mm:
* Source/WebCore/platform/win/BitmapInfo.h:
* Source/WebCore/platform/win/PopupMenuWin.cpp:
* Source/WebCore/platform/xr/openxr/OpenXRUtils.h:
* Source/WebCore/rendering/AttachmentLayout.mm:
* Source/WebCore/rendering/BorderPainter.cpp:
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
* Source/WebCore/rendering/RenderListBox.cpp:
* Source/WebCore/rendering/RenderMarquee.cpp:
* Source/WebCore/rendering/shapes/RasterShape.cpp:
* Source/WebCore/rendering/svg/SVGMarkerData.h:
* Source/WebCore/rendering/svg/SVGTextLayoutEngineBaseline.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleFontSizeFunctions.cpp:
* Source/WebCore/svg/SVGPathParser.cpp:
* Source/WebCore/svg/properties/SVGAnimationAdditiveValueFunctionImpl.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
* Source/WebKit/UIProcess/gtk/KeyBindingTranslator.cpp:
* Source/WebKit/UIProcess/gtk/ViewGestureControllerGtk.cpp:
* Source/WebKit/UIProcess/ios/WKKeyboardScrollingAnimator.mm:
* Source/WebKit/UIProcess/win/WebPopupMenuProxyWin.cpp:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
* Source/WebKitLegacy/ios/WebCoreSupport/WebFrameIOS.mm:
* Source/WebKitLegacy/mac/WebView/WebDynamicScrollBarsView.mm:
* Tools/TestRunnerShared/EventSerialization/mac/EventSerializerMac.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/GPUProcess.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
* Tools/TestWebKitAPI/Tests/ios/FullscreenTouchSecheuristicTests.cpp:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:

Canonical link: https://commits.webkit.org/261584@main
